src/query/catalog/src/table.rs

```diff
@@ -110,8 +110,13 @@ pub trait Table: Sync + Send {
         false
     }
 
-    /// Whether the table engine supports virtual columns optimization.
-    fn support_virtual_columns(&self) -> bool {
+    /// Whether the table engine supports virtual columns optimization for Variant types.
+    fn support_variant_virtual_columns(&self) -> bool {
```

Comment:
Virtual columns can be used not only for variant type, but also for internal columns of Map type or other nested types.

Reply:
I want to distinguish row_id and other virtual columns.

Reply:
To distinguish it from a `VirtualColumn`, the `row_id` can be treated as an `InternalColumn`, for example in the definition of `ColumnEntry`. https://github.com/datafuselabs/databend/blob/main/src/query/sql/src/planner/metadata.rs#L413